controller/src/beerocks/master/db/node.h

```diff
@@ -183,6 +187,27 @@ class node {
     };
     std::shared_ptr<radio> hostap;
 
+    class link_metrics_data{
+        public:
+        
+        link_metrics_data(){};
+        ~link_metrics_data(){};
+        
+        std::vector<ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo> transmitterLinkMetrics;
+        std::vector<ieee1905_1::tlvReceiverLinkMetric::sInterfacePairInfo> receiverLinkMetrics;
```

Comment:
I looked at the comment, and I disagreed with it.
We should choose std::vector over std::list if it is important for us to access an element in O(1) on non-sequential access, meaning access vector[0] and right away to access vector[5] for example. As well as to have the option to grow.
What you did here is to push elements one at the time which causes memory reallocation.
You could have used the vector constructor which takes iterator to the beginning and iterator to the end to save reallocation. But still, if you will erase elements you will still cause to reallocations.
Anyway, the erasing of elements does not exist and should be added.
@arnout

Reply:
i choose vector because i agree with @arnout that most of the times there will be only one element to this vector and a list is an overkill.
If we will need the ability to erase only one element in the future i'll add it.